neqo-http3/src/recv_message.rs

```diff
@@ -377,10 +379,10 @@ impl RecvMessage {
         }
     }
 
-    fn sanitize_headers(&self, headers: &mut Vec<Header>) -> Res<()> {
+    fn headers_valid(&self, headers: &[Header]) -> Res<()> {
         let mut method_value: Option<&String> = None;
         let mut pseudo_state = 0;
-        for header in headers.iter() {
+        for header in headers {
             let is_pseudo = Self::track_pseudo(&header.0, &mut pseudo_state, &self.message_type)?;
```

Comment:
Maybe for later, but large chunks of this function can now move to Header.  Think about a function that returns `Res<enum {
  Pseudo(MessageType, enum { Method, Authority, ... } ),
  Regular,
}>`